tests/test_codegen_advanced.cpp

```
@@ -251,6 +251,91 @@ TEST_F(CodeGenTest, LambdaArgTypeError) {
     EXPECT_THROW(runSource(src), std::runtime_error);
 }
 
+// ===== Effectively final capture tests =====
+
+TEST_F(CodeGenTest, CapturedVarReassignError) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)